engine/src/main/java/org/terasology/engine/core/module/ModuleManager.java

```diff
@@ -73,17 +73,32 @@ public ModuleManager(String masterServerAddress, List<Class<?>> classesOnClasspa
 
         registry = new TableModuleRegistry();
 
-        engineModule = loadEngineModule(classesOnClasspathsToAddToEngine);
+        engineModule = moduleFactory.createPackageModule("org.terasology.engine");
+        Module nui = createSyntheticPackageModule("nui", "3.0.0", "org.terasology.nui");
+        Module discordSubsystem = moduleFactory.createPackageModule("org.terasology.subsystem.discordrpc");
 
-        loadModulesFromApplicationPath(pathManager);
+        createDependency(engineModule, nui);
+
+        registry.add(nui);
+        registry.add(engineModule);
+        registry.add(discordSubsystem);
```

Comment:
hrmmm. Not sure about adding hardcoded references to subsystems in engine's ModuleManager.

The point of extracting subsystems—as I understand it—is to make it so the engine _doesn't_ always need to depend on them. i.e. a facade should be able to start an engine that doesn't have a DiscordRPC subsystem at all.